modules/adfs/lib/IdP/ADFS.php

```diff
@@ -163,19 +171,25 @@ public static function sendResponse(array $state)
         $idpEntityId = $idpMetadata->getString('entityid');
 
         $idp->addAssociation(array(
-            'id' => 'adfs:' . $spEntityId,
+            'id' => 'adfs:'.$spEntityId,
             'Handler' => 'sspmod_adfs_IdP_ADFS',
             'adfs:entityID' => $spEntityId,
         ));
 
-        $response = sspmod_adfs_IdP_ADFS::generateResponse($idpEntityId, $spEntityId, $nameid, $attributes);
+        $assertionLifetime = $spMetadata->getInteger('assertion.lifetime', null);
+        if ($assertionLifetime === null) {
+            $assertionLifetime = $idpMetadata->getInteger('assertion.lifetime', 300);
+        }
+
+        $response = sspmod_adfs_IdP_ADFS::generateResponse($idpEntityId, $spEntityId, $nameid, $attributes, $assertionLifetime);
 
         $privateKeyFile = \SimpleSAML\Utils\Config::getCertPath($idpMetadata->getString('privatekey'));
         $certificateFile = \SimpleSAML\Utils\Config::getCertPath($idpMetadata->getString('certificate'));
         $wresult = sspmod_adfs_IdP_ADFS::signResponse($response, $privateKeyFile, $certificateFile);
 
         $wctx = $state['adfs:wctx'];
-        sspmod_adfs_IdP_ADFS::postResponse($spMetadata->getValue('prp'), $wresult, $wctx);
+        $wreply = $state['adfs:wreply'] ? : $spMetadata->getValue('prp');
```

Comment:
`? :` that doesn't look right.

Reply:
It actually is. It's commonly known as the [_elvis operator_](http://php.net/manual/en/language.operators.comparison.php#language.operators.comparison.ternary).

Reply:
That's perfectly valid PHP??
"(expression1) ?: expression2 will return expression1 if it evaluates to true or expression2 otherwise."

Reply:
Could've used `||` too, I guess it's just a matter of coding style...

Reply:
Obviously I know about the ternary operator, but I was not aware of this "improvement" since PHP 5.3 where you can leave out the middle part.